src/pages/stacks-blockchain/sending-tokens.md

```diff
@@ -0,0 +1,207 @@
+const senderKey = createStacksPrivateKey(key);
+```
+
+-> Note: The code above also imports methods required for the next steps, including API configuration for the client library usage.
+
+## Step 3: Generating transaction
+
+To generate a token transfer transaction, we will be using the `makeSTXTokenTransfer()` transaction builder function:
+
+```js
+const recipient = 'SP3FGQ8Z7JY9BWYZ5WM53E0M9NK7WHJF0691NZ159';
+
+// amount of Stacks tokens to send (in microstacks). 1,000,000 microstacks are worth 1 STX token
+const amount = new BN(1000000);
+
+// skip automatic fee estimation
+const fee = new BN(0);
+
+// skip automatic nonce lookup
+const nonce = new BN(0);
+
+// override default setting to broadcast to the Testnet network
+const network = new StacksTestnet();
+
+const memo = 'hello world';
+
+const txOptions = {
+  recipient,
+  amount,
+  fee,
+  nonce,
+  senderKey: privateKeyToString(senderKey),
+  network,
+  memo,
+};
+
+...
+
+const transaction = await makeSTXTokenTransfer(txOptions);
+```
+
+The generation method will need a few more pieces of information, as specified in the `txOptions` object:
+
+| Parameter   | Description                                                                                                                      | Optional |
+| ----------- | -------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| `recipient` | The recipient Stacks address in c32check format                                                                                  | **No**   |
+| `amount`    | The amount of Stacks tokens to send denominated in microstacks                                                                   | **No**   |
+| `fee`       | The fee that the sender is willing to pay for miners to process the transaction. Denominated in microstacks                      | Yes      |
+| `nonce`     | A nonce is an integer that needs to be incremented by 1 for each sequential transaction from the same account. Nonces start at 0 | Yes      |
+| `senderKey` | A private key object                                                                                                             | Yes      |
+| `network`   | Specifies whether the transaction is meant for Stacks Mainnet or Testnet                                                         | Yes      |
+| `memo`      | A memo string to attach additional information to the transaction. This data is limited to 33 bytes                              | Yes      |
+
+### Estimating fees
+
+If not specified, the transaction builder will automatically estimate the fee. Estimated fee rate is supplied by a Stacks node so network access is required. The fee is calculated based on the estimate fee rate and the size of the transaction in bytes.
+
+Another way to estimate the fee is to use the `estimateTransfer()` function after you have constructed a transaction:
+
+```js
+estimateTransfer(transaction);
```

Comment:
Is this a function with side-effects? That is, it updates the `fee` value within the `transaction`? If yes, probably out of scope for this PR, but 1) I'd generally argue against functions with side-effects -- they are error-prone. and 2) the name should reflect what the function does. It's not just estimating, it's actually estimating _and_ setting the fee.

cc/ @yknl

Reply:
I just checked the [source](https://github.com/blockstack/stacks-transactions-js/blob/master/src/builders.ts#L504) and it returns just the fee, no side-effects. The code example should be better, ie setting a fee var and update the tx